### ISSUE TITLE
Admin install for optional workloads

### ISSUE THREAD

**Issue by marcpopMSFT** [dotnet/sdk#14956]

**Comment by marcpopMSFT** [comment]
@joeloff We had a review of this with the project system team.  One item to investigate are around how In-product Acquisition will work with this.  Presumably we return a list of individual component IDs and project system will trigger IPA for that but we need to confirm that VS setup supports that as that's different than how project system triggers IPA today (today they do it on project load for specific project type GUIDS).

Another was finding out how Codespaces will work wrt IPA if they components are not pre-configured on the Codespaces server.

Separately, we do need to investigate the project system experience for Xamarin projects but we can take that offline.

**Comment by joeloff** [comment]
Yup, IPA in CodeSpaces might be tricky. The idea with CodeSpaces generally is that everything you need is already installed. I'll need to check with VS folks how they handle optional workloads, because it will likely need to perform a machine wide install to modify VS setup.

The idea I originally had was to add the classifiers to the workloads for .NET Core. It's possible to express something like "when you open a .csproj that contains a property named X and its value is 17 then install something". I've done something similar using a project type GUID + component (without needing project system changes), so I was going along this route.

I'm hopeful to have a proof of concept ready soon

**Comment by marcpopMSFT** [comment]
I added a couple of checkboxes above to track these two issues.  For the second in particular, the project system team is very interested to know what work they may or may not have as for them, the existing way they trigger IPA is based on project type and so won't work for optional workloads as is.

**Comment by wli3** [comment]
@marcpopMSFT note _(stand alone installer)Exe/msi. Convert CLI command to msi command. Muti-targeted library._ is now committed after today's meeting. Since Blazor desktop would want this.
